nototools/android_patches.py

```diff
@@ -0,0 +1,431 @@
+        path.join(outdir, font_name),
+        exclude=EMOJI)
+
+
+def _fix_arabic_ligatures():
+  """Fix U+FDF2 and Allah-related ligatures in Naskh fonts.
+
+  U+FDF2 ARABIC LIGATURE ALLAH ISOLATED FORM missed the initial alef in
+  its glyph, while the Allah ligatures were incorrectly formed.
+
+  (8ae3a28c050ef9)"""
+
+  # noto-fonts/issues/384 was fixed, so this patch is not needed.
+  pass
+
+
+# below are used by _subset_symbols
+
+# Unicode blocks that we want to include in the font
+BLOCKS_TO_INCLUDE = """
+20D0..20FF; Combining Diacritical Marks for Symbols
+2100..214F; Letterlike Symbols
+2190..21FF; Arrows
+2200..22FF; Mathematical Operators
+2300..23FF; Miscellaneous Technical
+2400..243F; Control Pictures
+2440..245F; Optical Character Recognition
+2460..24FF; Enclosed Alphanumerics
+2500..257F; Box Drawing
+2580..259F; Block Elements
+25A0..25FF; Geometric Shapes
+2600..26FF; Miscellaneous Symbols
+2700..27BF; Dingbats
+27C0..27EF; Miscellaneous Mathematical Symbols-A
+27F0..27FF; Supplemental Arrows-A
+2800..28FF; Braille Patterns
+2A00..2AFF; Supplemental Mathematical Operators
+"""
+
+# One-off characters to be included, needed for backward compatibility and
+# supporting various character sets, including ARIB sets and black and white
+# emoji
+ONE_OFF_ADDITIONS = {
+    0x27D0, # ⟐ WHITE DIAMOND WITH CENTRED DOT
+    0x2934, # ⤴ ARROW POINTING RIGHTWARDS THEN CURVING UPWARDS
+    0x2935, # ⤵ ARROW POINTING RIGHTWARDS THEN CURVING DOWNWARDS
+    0x2985, # ⦅ LEFT WHITE PARENTHESIS
+    0x2986, # ⦆ RIGHT WHITE PARENTHESIS
+    0x2B05, # ⬅ LEFTWARDS BLACK ARROW
+    0x2B06, # ⬆ UPWARDS BLACK ARROW
+    0x2B07, # ⬇ DOWNWARDS BLACK ARROW
+    0x2B24, # ⬤ BLACK LARGE CIRCLE
+    0x2B2E, # ⬮ BLACK VERTICAL ELLIPSE
+    0x2B2F, # ⬯ WHITE VERTICAL ELLIPSE
+    0x2B56, # ⭖ HEAVY OVAL WITH OVAL INSIDE
+    0x2B57, # ⭗ HEAVY CIRCLE WITH CIRCLE INSIDE
+    0x2B58, # ⭘ HEAVY CIRCLE
+    0x2B59, # ⭙ HEAVY CIRCLED SALTIRE
+}
+
+# letter-based characters, provided by Roboto
+LETTERLIKE_CHARS_IN_ROBOTO = {
+    0x2100, # ℀ ACCOUNT OF
+    0x2101, # ℁ ADDRESSED TO THE SUBJECT
+    0x2103, # ℃ DEGREE CELSIUS
+    0x2105, # ℅ CARE OF
+    0x2106, # ℆ CADA UNA
+    0x2109, # ℉ DEGREE FAHRENHEIT
+    0x2113, # ℓ SCRIPT SMALL L
+    0x2116, # № NUMERO SIGN
+    0x2117, # ℗ SOUND RECORDING COPYRIGHT
+    0x211E, # ℞ PRESCRIPTION TAKE
+    0x211F, # ℟ RESPONSE
+    0x2120, # ℠ SERVICE MARK
+    0x2121, # ℡ TELEPHONE SIGN
+    0x2122, # ™ TRADE MARK SIGN
+    0x2123, # ℣ VERSICLE
+    0x2125, # ℥ OUNCE SIGN
+    0x2126, # Ω OHM SIGN
+    0x212A, # K KELVIN SIGN
+    0x212B, # Å ANGSTROM SIGN
+    0x212E, # ℮ ESTIMATED SYMBOL
+    0x2132, # Ⅎ TURNED CAPITAL F
+    0x213B, # ℻ FACSIMILE SIGN
+    0x214D, # ⅍ AKTIESELSKAB
+    0x214F, # ⅏ SYMBOL FOR SAMARITAN SOURCE
+}
+
+# default emoji characters in the BMP, based on
+# http://www.unicode.org/draft/Public/emoji/1.0/emoji-data.txt
+# We exclude these, so we don't block color emoji.
+BMP_DEFAULT_EMOJI = {
```

Comment:
OK.  Not quite sure what BMP is meant to imply, it doesn't seem to require a limit to BMP chars.